pkg/shp/cmd/build/run.go

```diff
@@ -71,6 +81,49 @@ func (c *RunCommand) Run(params *params.Params, ioStreams *genericclioptions.IOS
 		return err
 	}
 
-	fmt.Fprintf(ioStreams.Out, "BuildRun created %q\n", buildRun.Name)
+	if !c.Follow {
+		fmt.Fprintf(ioStreams.Out, "BuildRun created %q\n", buildRun.Name)
+		return nil
+	}
+
+	err := wait.PollImmediate(1*time.Second, 30*time.Second, func() (bool, error) {
+		if err := brr.Get(c.cmd.Context(), buildRun.Name, buildRun); err != nil {
+			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q not available: %q\n", buildRun.Name, err.Error())
+			return false, nil
+		}
+
+		if buildRun.Status.LatestTaskRunRef == nil {
+			fmt.Fprintf(ioStreams.ErrOut, "BuildRun %q created but cannot follow logs since there is no TaskRun reference\n", buildRun.Name)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+
+	tparams := &tkncli.TektonParams{}
+	tparams.SetNamespace(buildRun.Namespace)
+	opts := &tknopt.LogOptions{
```

Comment:
This sounds really similar to to Pod's GetLogs (with `follow` option enabled), so I wonder what are the advantages of pulling in Tekton CLI instead of using it directly. Do you have a specific reason for this move?

Reply:
Yeah I looked at the more vanilla k8s pod get logs option, but the tkn form had additional code implementation on top of it that I was headed down the road of implementing anyway.

Maximum code reuse won out in the end in my mind.

And I'm also of the belief that this will not be the first case that we can leverage work already done in `tkn`

Reply:
I also tried the log streaming own my own, and I came up with the following:
- https://github.com/otaviof/cli/blob/b70442b9b44a3d3b41a33d3841a99c1ecced85e9/pkg/shp/tail/tail.go
- https://github.com/otaviof/cli/blob/b70442b9b44a3d3b41a33d3841a99c1ecced85e9/pkg/shp/tail/tail_test.go

And then in combination with a reactor for pod state changes, it can stream the logs.
- https://github.com/otaviof/cli/blob/b70442b9b44a3d3b41a33d3841a99c1ecced85e9/pkg/shp/reactor/pod_watcher.go
- https://github.com/otaviof/cli/blob/b70442b9b44a3d3b41a33d3841a99c1ecced85e9/pkg/shp/reactor/pod_watcher_test.go

Do you think we could reuse those, and also reusing the unit-tests?

Reply:
@otaviof the unit tests minimally are a great starting point for me to try an incorporate into the existing paths I have ... most notably the pod watcher stuff maps well to the buildrun watch I added based on your prior comment

I'll start there and once I get a handle on it, see what if any of the tail stuff applies to the code I'm reusing from `tkn`